raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -1170,6 +1170,144 @@ def test_init_mediator():
+    )
+    from_route = factories.route_from_channel(from_channel)
+
+    from_transfer = factories.make_signed_transfer_for(
+        channel_state=from_channel,
+        amount=amount,
+        initiator=HOP1,
+        target=target,
+        expiration=from_expiration,
+        secret=EMPTY_HASH,
+        allow_invalid=True,
+    )
+
+    channel1 = factories.make_channel(
+        our_balance=amount,
+        partner_address=HOP2,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    available_routes = [factories.route_from_channel(channel1)]
+    channel_map = {
+        from_channel.identifier: from_channel,
+        channel1.identifier: channel1,
+    }
+
+    block_number = 1
+    init_state_change = ActionInitMediator(
+        available_routes,
+        from_route,
+        from_transfer,
+    )
+
+    mediator_state = None
+    iteration = mediator.state_transition(
+        mediator_state,
+        init_state_change,
+        channel_map,
+        pseudo_random_generator,
+        block_number,
+    )
+
+    assert not iteration.new_state
+
+
+def test_mediator_secret_reveal_empty_hash():
+    amount = UNIT_TRANSFER_AMOUNT
+    target = HOP2
+    from_expiration = UNIT_SETTLE_TIMEOUT - UNIT_REVEAL_TIMEOUT
+    pseudo_random_generator = random.Random()
+
+    from_channel = factories.make_channel(
+        partner_balance=amount,
+        partner_address=UNIT_TRANSFER_SENDER,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    from_route = factories.route_from_channel(from_channel)
+
+    from_transfer = factories.make_signed_transfer_for(
+        from_channel,
+        amount,
+        HOP1,
+        target,
+        from_expiration,
+        UNIT_SECRET,
+    )
+
+    channel1 = factories.make_channel(
+        our_balance=amount,
+        partner_address=HOP2,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    available_routes = [factories.route_from_channel(channel1)]
+    channel_map = {
+        from_channel.identifier: from_channel,
+        channel1.identifier: channel1,
+    }
+
+    block_number = 1
+    init_state_change = ActionInitMediator(
+        available_routes,
+        from_route,
+        from_transfer,
+    )
+
+    mediator_state = None
+    iteration = mediator.state_transition(
+        mediator_state,
+        init_state_change,
+        channel_map,
+        pseudo_random_generator,
+        block_number,
+    )
+    assert isinstance(iteration.new_state, MediatorTransferState)
+    assert iteration.new_state.transfers_pair[0].payer_transfer == from_transfer
+    current_state = iteration.new_state
+
+    # an empty hash offchain secret reveal should be rejected
+    receive_secret = ReceiveSecretReveal(
+        EMPTY_HASH,
+        HOP2,
+    )
+    iteration = mediator.state_transition(
+        current_state,
+        receive_secret,
+        channel_map,
+        pseudo_random_generator,
+        block_number + 1,
+    )
+    assert len(iteration.events) == 0
```

Comment:
I would also check for the lock here... reason being that at some point we might change lock related things and break this... so this will be a good defensive check.

Reply:
same problem as above